core/mutator/mutator.go

```diff
@@ -58,18 +58,14 @@ type Mutator interface {
 
 // MutationStorage reads and writes mutations to the database.
 type MutationStorage interface {
-	// ReadRange reads all mutations for a specific given mapID and sequence
-	// range. The range is identified by a starting sequence number and a
-	// count. Note that startSequence is not included in the result.
-	// ReadRange stops when endSequence or count is reached, whichever comes
-	// first. ReadRange also returns the maximum sequence number read.
-	ReadRange(ctx context.Context, mapID int64, startSequence, endSequence uint64, count int32) (uint64, []*pb.EntryUpdate, error)
-	// ReadAll reads all mutations starting from the given sequence number.
-	// Note that startSequence is not included in the result. ReadAll also
-	// returns the maximum sequence number read.
-	// ReadAll will not return more than count entries.
-	ReadAll(ctx context.Context, mapID int64, startSequence uint64, count int) (uint64, []*pb.EntryUpdate, error)
+	// ReadPage returns mutations in the interval (start, end] for mapID.
+	// pageSize specifies the maximum number of items to return.
+	// Returns the maximum sequence number returned.
```

Comment:
If pageSize were left to the implementation then the client would be able to determine if truncation had occured, right?  Since the returned maxSeqNum would be < end

Reply:
This API is used to paginate through the mutations in an epoch. As a result, the client doesn't really know `start` or `end`, which makes setting a `pageSize` useful.

Reply:
OK, fair enough.